service/kvmconfig/v2/resource_set.go

```diff
@@ -95,6 +96,19 @@ func NewResourceSet(config ResourceSetConfig) (*framework.ResourceSet, error) {
 		}
 	}
 
+	var clusterRoleBinding framework.Resource
+	{
+		c := clusterrolebinding.DefaultConfig()
+
+		c.K8sClient = config.K8sClient
+		c.Logger = config.Logger
+
+		clusterRoleBinding, err = clusterrolebinding.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
```

Comment:
If we want to enable RBAC (strictly, now it has `AlwaysAllow` mode too) in all installations then cluster role bindings are needed, otherwise deployments will not work anymore(no permissions).
There two options:
- Wait all clusters move to the latest version and then enable RBAC
- Modify frozen version 

The last one is easier because we don't have to track clusters to know when we can change to RBAC. But I don't want to break our upgrade system, so let's close this PR and approve [v3](#349).
The plan for the rolls out is described [here](https://github.com/giantswarm/giantswarm/issues/1912#issuecomment-360109452) Feel free to add feedback

Reply:
I just say this again, when we modify a frozen version we make a joke of ourselves. It is also that the already running guest clusters are not updated automatically. So we can violate our own version rules and processes but this will still not bring RBAC to the guest clusters. We would need to update them all manually one by one. I am not sure this is what we want. In your plan it is even stated to patch manually or wait until they are migrated. What is it now? I still think this here is wrong for all the reasons why we came up with our versioning concepts.

Reply:
Maybe I am wrong but I thought versioning was more focused on guest clusters than host. Anyway, I would like you can guide me a bit how it can be improved, obviously, you have more thoughts and experience with it. Really, I don't want to mess up with the versioning so let's follow the process and only add the changes on the v3.

About the patching on the existing deployments, how it can be done automatically? 
My first impression was in the update process the [deployments compare desired and current of the entire `Spec` section](https://github.com/giantswarm/kvm-operator/blob/master/service/kvmconfig/v3/resource/deployment/update.go#L108) so the operator will update existing deployments with the service account name. But it does not work like that. AFAIK there is a reconciliation loop trigger from time to time and they will be able to update it properly. Why this not happen?

Reply:
> About the patching on the existing deployments, how it can be done automatically?

Not at all right now. This is why we work on the update story to automate it. Right now you have to drain and pods roll deployments in KVM manually. We did this already but it is such a waste of time and there should be a really good reason to do it because it is manual and we can just easily fuck up everything which is why I strong discourage to do it. If we can get around it in a different way I would always prefer the different way. 


> My first impression was in the update process the deployments compare desired and current of the entire Spec section so the operator will update existing deployments with the service account name.

Please see https://github.com/giantswarm/kvm-operator/blob/bf5c288c5610572bcf3f096b592c63821e7282fe/service/kvmconfig/v3/resource/deployment/update.go#L76. Deployment updates are disabled everywhere because they are not graceful. If we would enable it right now we would screw all customer workloads and probably lose all of our customers because of it. I am currently working on  https://github.com/giantswarm/node-operator in order to make upgrades graceful. Once this is in place we can wire and enable updates in KVM. This will still take a bit of time though.

Reply:
ok I see

>  If we can get around it in a different way I would always prefer the different way.
The other way is doing something like (bash pseudocode)
```
clusters=$(kubectl get kvmconfigs -n default)
for cluster in $clusters do
   deployments=$(kubectl get deploys -n $cluster)
   for deployment in deployments do
     kubectl patch deployment $deployment --patch "...serviceAccountName: $cluster..."
  done
done
```
It should be well tested before, but then a installation can be patched for existing guest cluster deployments. WDYT?

Reply:
I don't know. I think patching a deployment means rolling the pods. If we apply RBAC rules the pods should definitely be rolled because otherwise the rules are not applied to the workloads. When we roll pods we have the overhead of the manual draining and rolling. Discuss this with your team and find a way to move forward. I just want to make sure we comply with our own rules and that we do not screw with running customer workloads.

Reply:
Yes, you are right, pods are rolled out :(. I have thought that for a service account name addition it would not be the case, but I have just tested. Then we need another solution. I will discuss with my team, but IMO wait for all clusters are updated is better as RBAC is not something super urgent

Anyway, these v3 changes can be rolled out since not affect existing clusters for now. (I have corrected the DefaultConfig ;) )

Thanks for the discussion, really thankful for your insight

Reply:
No worries buddy. You are welcome.